### PR TITLE
Add IPv6 support

### PR DESCRIPTION
**Pull request by globin**
Adds support to listen on IPv6 (and IPv4, this should be the default in the docs after this is merged):
```
!plugin config Webserver {'HOST': '::', 'PORT': 3141}
```

Uses `getaddrinfo` and `AF_UNSPEC` to support both IP4 and IP6.

### CODE REVIEW

**Comment by globin** [comment]
ping @gbin

**Comment by gbin-argo** [comment]
Thanks

File: rocket/listener.py [hunk 1]
```diff
@@ -65,38 +58,53 @@ def __init__(self, interface, queue_size, active_queue, *args, **kwargs):
...
-            msg = "Cannot set TCP_NODELAY, things might run a little slower"
-            self.err_log.warning(msg)
+        listener = None
+        for res in socket.getaddrinfo(self.addr, self.port, socket.AF_UNSPEC,
+                                      socket.SOCK_STREAM, 0, socket.AI_PASSIVE):
+            af, socktype, proto, canonname, sa = res
+            try:
+                listener = socket.socket(af, socktype, proto)
+            except OSError as msg:
+                listener = None
```
Comment (gbin-argo) [inline rocket/listener.py:68]: Shouldn't we at least log a message or the exception?

File: rocket/listener.py [hunk 2]
```diff
@@ -65,38 +58,53 @@ def __init__(self, interface, queue_size, active_queue, *args, **kwargs):
...
+                listener.close()
+                msg = "Socket %s in use by other process and it won't share."
+                self.err_log.error(msg % sa)
+                listener = None
+                continue
+            break
+
+        # no socket found
+        if not listener:
+            self.err_log.error("failed to get socket.")
```
Comment (gbin-argo) [inline rocket/listener.py:108]: Because this won't leave any specific explanation

**Review by gbin-argo — CHANGES_REQUESTED** [review]

**Comment by gbin** [comment]
Just logging the exception correctly and I'll merge this one.

**Comment by globin** [comment]
Done

**Review by gbin — APPROVED** [review]

**Pull request merged**